sw/device/lib/crypto/impl/kdf.c

```diff
@@ -35,40 +35,62 @@ otcrypto_status_t otcrypto_kdf_kmac(
     return OTCRYPTO_BAD_ARGS;
   }
 
-  // Check the private key checksum.
-  if (integrity_blinded_key_check(&key_derivation_key) != kHardenedBoolTrue) {
+  // Check for null label with nonzero length.
+  if (kdf_label.data == NULL && kdf_label.len != 0) {
+    return OTCRYPTO_BAD_ARGS;
+  }
+  // Because of KMAC HWIPs prefix limitation, `label` should not exceed
+  // `kKmacCustStrMaxSize` bytes.
+  if (kdf_label.len > kKmacCustStrMaxSize) {
     return OTCRYPTO_BAD_ARGS;
   }
 
-  // The underlying KMAC implementation is not hardened yet.
-  if (key_derivation_key.config.security_level !=
-          kOtcryptoKeySecurityLevelLow ||
-      keying_material->config.security_level != kOtcryptoKeySecurityLevelLow) {
-    return OTCRYPTO_NOT_IMPLEMENTED;
+  // Check for null context with nonzero length.
+  if (kdf_context.data == NULL && kdf_context.len != 0) {
+    return OTCRYPTO_BAD_ARGS;
+  }
+
+  // Check the private key checksum.
+  if (integrity_blinded_key_check(&key_derivation_key) != kHardenedBoolTrue) {
+    return OTCRYPTO_BAD_ARGS;
   }
 
   // Check `key_len` is supported by KMAC HWIP.
   // The set of supported key sizes is {128, 192, 256, 384, 512).
   HARDENED_TRY(kmac_key_length_check(key_derivation_key.config.key_length));
 
+  kmac_blinded_key_t kmac_key = {
+      .share0 = NULL,
+      .share1 = NULL,
+      .hw_backed = key_derivation_key.config.hw_backed,
+      .len = key_derivation_key.config.key_length,
+  };
   // Validate key length of `key_derivation_key`.
   if (key_derivation_key.config.hw_backed == kHardenedBoolTrue) {
-    return OTCRYPTO_NOT_IMPLEMENTED;
+    // Check that 1) key size matches sideload port size, 2) keyblob length
+    // matches diversification length.
+    if (keyblob_share_num_words(key_derivation_key.config) * sizeof(uint32_t) !=
+            kKmacSideloadKeyLength / 8 ||
+        key_derivation_key.keyblob_length != 8 * sizeof(uint32_t)) {
```

Comment:
I don't think you need this check; it's included in `keyblob_to_keymgr_diversification`:
https://github.com/lowRISC/opentitan/blob/deb7ee190f923d143f14c07db5f5f2604242e675/sw/device/lib/crypto/impl/keyblob.c#L127